lib/tree.rb

```diff
@@ -1,38 +1,55 @@
 class NoApplesError < StandardError; end
 
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @height = 0
+    @age = 0
+    @apples = 0
+    @alive = true
   end
 
   def age!
+    @age+=1
+    @alive = false unless @age < 9
+    @height+=rand(20)
+    self.add_apples
```

Comment:
The `self` shouldn't be necessary here - `self` is the default receiver for every command in ruby.

Reply:
Joe is right. The only time `self.` is needed is calling a setter method, like `self.foo = "bar"`. Ruby will assume `foo = "bar"` is local variable assignment instead of calling the `foo=` method.